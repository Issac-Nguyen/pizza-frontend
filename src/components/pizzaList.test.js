import React from 'react'
import {render, cleanup} from '@testing-library/react'
import PizzaList from './pizzaList'

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });

jest.mock('./pizza', () => () => (
    <div>pizza</div>
))

describe('PizzaList', () => {
    afterEach(cleanup);

    it('should take a snapshot with pizza has no length', () => {
        const { asFragment } = render(<PizzaList />);
        
        expect(asFragment(<PizzaList />)).toMatchSnapshot()
     })

     it('should take a snapshot with pizza has length', () => {
        const { asFragment } = render(<PizzaList pizzas={[{id: 1}]}/>);
        
        expect(asFragment(<PizzaList />)).toMatchSnapshot()
     })

     it('should have item pizza when pizza has length', () => {
        const { container } = render(<PizzaList pizzas={[{id: 1}]}/>);
        expect(container.querySelector('.pizza-area')).toHaveTextContent('pizza')
     })
})