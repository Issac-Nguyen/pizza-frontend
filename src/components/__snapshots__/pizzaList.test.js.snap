// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaList should take a snapshot with pizza has length 1`] = `
<DocumentFragment>
  <div
    class="ant-row pizza-area"
  >
    <div>
      pizza
    </div>
  </div>
</DocumentFragment>
`;

exports[`PizzaList should take a snapshot with pizza has no length 1`] = `
<DocumentFragment>
  <div
    class="ant-row pizza-area"
  />
</DocumentFragment>
`;
